Update animals table

samuelmoiwa opened the pull request:
## Vet Clinic Database

## Solution description

In this project, I used the relational database to create the initial data structure for a Vet clinic.

> I created a table to store information about the various animals.
> I inserted data into the table
> I added queries

## Address affected and ensured
* data.sql
* queries.sql
* schema.sql
---

## Screenshot of the queries:


![Screenshot 2022-12-13 200152](https://user-images.githubusercontent.com/38554511/207433047-9947ed17-a2b1-4e4b-8664-952d6440a471.png)

samuelmoiwa committed:
Update queries.sql

samuelmoiwa committed:
Update data.sql

Davidosky007 commented on queries.sql:
```diff
@@ -8,3 +8,35 @@ SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
+
+
+BEGIN;
+UPDATE animals SET species = 'unspecified';
```
* Kindly verify that the change was made after `species = unspecified` and verify that the species columns went back to the state before the transaction.
 ![image](https://user-images.githubusercontent.com/52497246/207435729-983980bd-510a-4292-af5a-23e563a6981a.png)

Davidosky007 commented on queries.sql:
```diff
@@ -8,3 +8,35 @@ SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
+
+BEGIN;
+UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon';
+UPDATE animals SET species = 'pokemon' WHERE species IS NULL;
```
* Kindly verify that the change was made  here as requested in the[ project requirement](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/vet_clinic_query_update_tables.md#:~:text=Verify%20that%20change%20was%20made%20and%20persists%20after%20commit.)

Davidosky007 commented on queries.sql:
```diff
@@ -8,3 +8,35 @@ SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
+BEGIN;
+UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon';
+UPDATE animals SET species = 'pokemon' WHERE species IS NULL;
+COMMIT;
```
* Kindly `verify that change persists after commit`

Davidosky007 reviewed (changes requested):
Hi @samuelmoiwa 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
## Highlights
* Well written PR description 👍 
* Proper use of Gitflow 👍 

## Required Changes ♻️
* Kindly Add a column species of type string to your animals table. Modify your schema.sql file. as stated in the [project requirement ](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/vet_clinic_query_update_tables.md#:~:text=Add%20a%20column%20species%20of%20type%20string%20to%20your%20animals%20table.%20Modify%20your%20schema.sql%20file.)
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidsoky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

samuelmoiwa committed:
Update queries.sql

addod19 reviewed (approved):
### Hi @samuelmoiwa, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

### Highlights
- [ ] Nice job fixing the comments from the previous code review :100: 


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

addod19 reviewed (changes requested):
### Hi @samuelmoiwa, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Highlights
- [ ] Nice job addressing some of the comments made by the previous code reviewer

 ## Required Changes ♻️
_Check the comments under the review._
- [x] It would be better if we fix our schema.sql file by altering our table and adding the `species` column to our table

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

samuelmoiwa commented:
> ### Hi @samuelmoiwa,
> Great work on making the changes requested by a previous reviewer 👏 You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.
> 
> ## Highlights
> * [ ]  Nice job addressing some of the comments made by the previous code reviewer
> 
> ## Required Changes ♻️
> _Check the comments under the review._
> 
> * [x]  It would be better if we fix our schema.sql file by altering our table and adding the `species` column to our table
> 
> ## Optional suggestions
> _Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._
> 
> Cheers and Happy coding!👏👏👏
> 
> Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.
> 
> **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
> 
> _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._ _Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Hello @addod19  ,
the instructions are not clear.
please help explain more clearly so I can understand what to change.
thanks ;)

samuelmoiwa committed:
Update schema.sql

danmainah commented on schema.sql:
```diff
@@ -6,5 +6,6 @@ CREATE TABLE animals (
  neutered BOOLEAN,
- weight_kg NUMERIC
+ weight_kg NUMERIC,
+ species TEXT
```
- [ ] @samuelmoiwa great work on implementing the changes requested, however, you didn't use the best practice to update the schema, you need to use the Alter keyword eg ` ALTER TABLE table-name ADD species varchar(255);`, hope the requirements are now clear, kindly contact me on slack if need of more help.
![image](https://user-images.githubusercontent.com/61408860/207528294-2cfab02f-6285-4674-81e7-e2f9d9b26a3a.png)

danmainah reviewed (changes requested):
Hi @samuelmoiwa , 

Great work on making the changes requested by a previous reviewer 👏  👏 👏
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
✔️ Gitflow was correctly used

 # Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

samuelmoiwa committed:
Update schema.sql

HAMMAS-SALEEM reviewed (approved):
# Status: Approved :green_circle: :1st_place_medal: :100:
![success](https://camo.githubusercontent.com/6ada59f6189497c1c2cae6d06f8154b7307a58643f232340e288140ed469eb28/68747470733a2f2f6d65646961322e67697068792e636f6d2f6d656469612f6c3366516333614356304871707a5763552f67697068792e676966)

### Hey @samuelmoiwa 👋

Great work on implementing the requested changes. Your project is complete. :blush:
There is nothing else to say about your project. :+1:

## Highlights :1st_place_medal:

✔️ Descriptive PR Title
✔️ Descriptive PR Summary
✔️ Follow Gitflow
✔️ Following Requirements

# Optional

_All the changes with optional tags are not crucial enough to prevent you from getting the approval but we highly recommend to students that they should take these changes into consideration._

Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

---------------------------------------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
